src/OneMightyRoar/PHP_ActiveRecord_Components/Cache/Adapter/DoctrineCacheAdapter.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * PHP-ActiveRecord-Components
+ *
+ * Useful common components for a php-activerecord based project
+ *
+ * @copyright   2013 One Mighty Roar
+ * @link        http://onemightyroar.com
+ */
+
+namespace OneMightyRoar\PHP_ActiveRecord_Components\Cache\Adapter;
+
+use Doctrine\Common\Cache\CacheProvider;
+use OneMightyRoar\PHP_ActiveRecord_Components\Cache\CacheAdapterInterface;
+
+/**
+ * DoctrineCacheAdapter
+ *
+ * An adapter to translate a Doctrine {@see CacheProvider} into a PHP-ActiveRecord
+ * compatible cache adapter, through the {@see CacheAdapterInterface}
+ *
+ * @package OneMightyRoar\PHP_ActiveRecord_Components\Cache\Adapter
+ */
+class DoctrineCacheAdapter implements CacheAdapterInterface
+{
+
+    /**
+     * Constants
+     */
+
+    /**
+     * The default value for the option of whether or not to flush the namespace
+     * instead of the entire cache for flush operations
+     *
+     * @type bool
+     */
+    const FLUSH_NAMESPACE_DEFAULT = false;
+
+
+    /**
+     * Properties
+     */
+
+    /**
+     * The Doctrine cache provider to be adapted
+     *
+     * @type CacheProvider
+     */
+    private $adapted_provider;
+
+    /**
+     * Whether or not to flush the namespace instead of the entire cache when
+     * committing a cache flush
+     *
+     * @type bool
+     */
+    private $flush_namespace = false;
```

Comment:
We don't need the default here. Could have defensively, but kind of confusing when we have a constant above called `FLUSH_NAMESPACE_DEFAULT `. If you change one, you gotta change the other? Minor, I know.